File: app/src/main/java/fr/paug/androidmakers/ui/adapter/ScheduleSession.java
package fr.paug.androidmakers.ui.adapter;

import androidx.annotation.NonNull;

import fr.paug.androidmakers.model.ScheduleSlot;
import fr.paug.androidmakers.util.EmojiUtils;

public class ScheduleSession implements Comparable<ScheduleSession> {

    private ScheduleSlot mScheduleSlot;
    private String mTitle;
    private String mLanguage;

    public ScheduleSession(ScheduleSlot scheduleSlot, String title, String language) {
        mScheduleSlot = scheduleSlot;
        mTitle = title;
        mLanguage = language;
    }

    public long getStartTimestamp() {
        return mScheduleSlot.startDate;
    }

    public long getEndTimestamp() {
        return mScheduleSlot.endDate;
    }

    public String getTitle() {
        return mTitle;
    }

    public String getLanguage() {
        return mLanguage;
    }

    public String getLanguageInEmoji() {
        return EmojiUtils.getLanguageInEmoji(mLanguage);
    }

    public int getRoomId() {
        return mScheduleSlot.room;
    }

    public int getSessionId() {
        return mScheduleSlot.sessionId;
    }

    @Override
    public String toString() {
        return "ScheduleSession{" +
                "mScheduleSlot=" + mScheduleSlot +
                ", mTitle='" + mTitle + '\'' +
                '}';
    }

    @Override
    public int compareTo(@NonNull ScheduleSession o) {
        return Long.valueOf(this.mScheduleSlot.startDate).compareTo(o.mScheduleSlot.startDate);
    }

}

File: app/src/main/java/fr/paug/androidmakers/ui/adapter/ScheduleSession.kt
package fr.paug.androidmakers.ui.adapter

import fr.androidmakers.store.model.ScheduleSlot
import fr.androidmakers.store.model.Speaker
import fr.paug.androidmakers.util.EmojiUtils

data class ScheduleSession(
    var slot: ScheduleSlot,
    val title: String = "",
    val language: String = "",
    val speakers: List<Speaker> = mutableListOf()
) : Comparable<ScheduleSession> {

  val languageInEmoji: String get() = EmojiUtils.getLanguageInEmoji(language) ?: ""

  val sessionId get() = slot.sessionId
  val roomId get() = slot.roomId

  override fun compareTo(other: ScheduleSession): Int {
    return slot.startDate.compareTo(other.slot.startDate)
  }

}